### PR TITLE
Fix Bidirection noise

### PR DESCRIPTION
**Pull request by best25792**

### CODE REVIEW

**Review by kkalavantavanich — CHANGES_REQUESTED** [review]
'n++' bug

**Comment by kkalavantavanich** [comment]
Delete main.cpp ???

File: _main.cpp [hunk 1]
```diff
@@ -20,137 +21,149 @@ const int32_t white_thresholds[4] = { 700, 700, 1050, 1270 };
...
+	if (!robot.Connect(Create_Comport)) {
 		cout << "Error : Can't connect to robot @" << Create_Comport << endl;
 		return -1;
 	}
 
 	robot.DriveDirect(0, 0);
 	cvNamedWindow("Robot");
 
-	int previous[4] = {0, 0, 0, 0};
+	int clint direction_checkediff_history[4] = { 0, 0, 0, 0 };
```
Comment (kkalavantavanich) [inline _main.cpp]: 'clint' -> cliff_history[4] & direction_checked ??

Comment (best25792) [reply _main.cpp]: ok

**Review by kkalavantavanich — APPROVED** [review]

**Pull request merged**